inc/classes/class-rocket-background-database-optimization.php

```diff
@@ -0,0 +1,171 @@
+					}
+
+					$this->count[ $item ] = $number;
+				}
+				break;
+			case 'auto_drafts':
+				$query = $wpdb->get_col( "SELECT ID FROM $wpdb->posts WHERE post_status = 'auto-draft'" );
+				if ( $query ) {
+					$number = 0;
+					foreach ( $query as $id ) {
+						$number += (int) wp_delete_post( intval( $id ), true );
+					}
+
+					$this->count[ $item ] = $number;
+				}
+				break;
+			case 'trashed_posts':
+				$query = $wpdb->get_col( "SELECT ID FROM $wpdb->posts WHERE post_status = 'trash'" );
+				if ( $query ) {
+					$number = 0;
+					foreach ( $query as $id ) {
+						$number += (int) wp_delete_post( $id, true );
+					}
+
+					$this->count[ $item ] = $number;
+				}
+				break;
+			case 'spam_comments':
+				$query = $wpdb->get_col( "SELECT comment_ID FROM $wpdb->comments WHERE comment_approved = 'spam'" );
+				if ( $query ) {
+					$number = 0;
+					foreach ( $query as $id ) {
+						$number += (int) wp_delete_comment( intval( $id ), true );
+					}
+
+					$this->count[ $item ] = $number;
+				}
+				break;
+			case 'trashed_comments':
+				$query = $wpdb->get_col( "SELECT comment_ID FROM $wpdb->comments WHERE (comment_approved = 'trash' OR comment_approved = 'post-trashed')" );
+				if ( $query ) {
+					$number = 0;
+					foreach ( $query as $id ) {
+						$number += (int) wp_delete_comment( intval( $id ), true );
+					}
+
+					$this->count[ $item ] = $number;
+				}
+				break;
+			case 'expired_transients':
+				$time = isset( $_SERVER['REQUEST_TIME'] ) ? (int) $_SERVER['REQUEST_TIME'] : time();
+				$query = $wpdb->get_col( "SELECT option_name FROM $wpdb->options WHERE option_name LIKE '_transient_timeout%' AND option_value < '$time'" );
+
+				if ( $query ) {
+					$number = 0;
+					foreach ( $query as $transient ) {
+						$key = str_replace( '_transient_timeout_', '', $transient );
+						$number += (int) delete_transient( $key );
+					}
+
+					$this->count[ $item ] = $number;
+				}
+				break;
+			case 'all_transients':
+				$query = $wpdb->get_col( "SELECT option_name FROM $wpdb->options WHERE option_name LIKE '_transient_%' OR option_name LIKE '_site_transient_%'" );
+				if ( $query ) {
+					$number = 0;
+					foreach ( $query as $transient ) {
+						if ( strpos( $transient, '_site_transient_' ) !== false ) {
+							$number += (int) delete_site_transient( str_replace( '_site_transient_', '', $transient ) );
+						} else {
+							$number += (int) delete_transient( str_replace( '_transient_', '', $transient ) );
+						}
+					}
+
+					$this->count[ $item ] = $number;
+				}
+				break;
+			case 'optimize_tables':
+				$query = $wpdb->get_results( "SELECT table_name, data_free FROM information_schema.tables WHERE table_schema = '" . DB_NAME . "' and Engine <> 'InnoDB' and data_free > 0" );
+				if ( $query ) {
+					$number = 0;
+					foreach ( $query as $table ) {
+						$number += (int) $wpdb->query( "OPTIMIZE TABLE '$table->table_name'" );
```

Comment:
No quotes around the database name here (`DB_NAME`), this will trigger a SQL error.